File: android/src/main/kotlin/com/superwall/superwallkit_flutter/bridges/ComputedPropertyRequestBridge.kt
package com.superwall.superwallkit_flutter.bridges

import ComputedPropertyRequest

fun ComputedPropertyRequest.toJson(): Map<String, Any?> {
    return mapOf(
        "type" to type.toJson(),
        "eventName" to eventName
    )
}

fun ComputedPropertyRequest.ComputedPropertyRequestType.toJson(): String {
    return when (this) {
        ComputedPropertyRequest.ComputedPropertyRequestType.MINUTES_SINCE -> "minutesSince"
        ComputedPropertyRequest.ComputedPropertyRequestType.HOURS_SINCE -> "hoursSince"
        ComputedPropertyRequest.ComputedPropertyRequestType.DAYS_SINCE -> "daysSince"
        ComputedPropertyRequest.ComputedPropertyRequestType.MONTHS_SINCE -> "monthsSince"
        ComputedPropertyRequest.ComputedPropertyRequestType.YEARS_SINCE -> "yearsSince"
    }
}

File: ios/Classes/Bridges/ComputedPropertyRequestBridge.swift
import Flutter
import SuperwallKit

extension ComputedPropertyRequest {
  func toJson() -> [String: Any] {
    var dict = [String: Any]()
    dict["type"] = type.toJson()
    dict["eventName"] = eventName
    return dict
  }
}

extension ComputedPropertyRequest.ComputedPropertyRequestType {
  func toJson() -> String {
    switch self {
      case .minutesSince:
        return "minutesSince"
      case .hoursSince:
        return "hoursSince"
      case .daysSince:
        return "daysSince"
      case .monthsSince:
        return "monthsSince"
      case .yearsSince:
        return "yearsSince"
    }
  }
}
